### DISCUSSION TITLE
[Feature]: Rename @remix-run/architect adapter to @remix-run/aws-lambda

### DISCUSSION THREAD

**Discussion by shamsup** [remix-run/remix#3905] [Proposals]
### What is the new or updated feature that you are suggesting?

1. I think it would be beneficial in the long run to rename this adapter to `@remix-run/aws-lambda` to avoid confusion. 
2. Remove the peer dependency on `@architect/architect` since it is not used

### Why should this feature be included?

1. The `@remix-run/architect` adapter is actually a generic adapter for AWS Lambda behind API Gateway. This package can be used as the adapter for projects using AWS CDK, AWS SAM, and serverless framework in addition to architect (and lambda without a framework).
2. Currently, the architect adapter has no need for a peer dependency of architect because it is not used within the package.

I think this will avoid confusion moving forward for people who want to run Remix within AWS Lambda but are using another framework to work with the service.

**Comment by GautierBlandin** [comment]
I agree with the sentiment, however I think it should be renamed to @remix-run/api-gateway-v2 (or @remix-run/api-gateway-http), because it's not a generic AWS solution, but aimed only at api-gateway HTTP. It would also be beneficial for the community to have other adapters for AWS entry points, such as api-gateway REST and lambda function URLs.

**Comment by gotgenes** [comment]
@wingleung published a third-party library with Lambda adapters called [remix-aws](https://github.com/wingleung/remix-aws). I think it addresses the need called out by this issue, in that, it is completely separated out from Architect.

Thanks to @MattyBalaam for [sharing the existence of this library in a separate discussion](https://github.com/remix-run/remix/discussions/8836#discussioncomment-9729752).
